cmd/promtool/main.go

```diff
@@ -747,3 +766,45 @@ func (j *jsonPrinter) printLabelValues(v model.LabelValues) {
 	//nolint:errcheck
 	json.NewEncoder(os.Stdout).Encode(v)
 }
+
+// BackfillRule backfills recording rules from the files provided. The output are blocks of data
+// at the outputDir location.
+func BackfillRule(url, start, end, outputDir string, evalInterval time.Duration, files ...string) int {
+	ctx := context.Background()
+	stime, etime, err := parseStartTimeAndEndTime(start, end)
+	if err != nil {
+		fmt.Fprintln(os.Stderr, err)
+		return 1
+	}
+	cfg := RuleImporterConfig{
+		Start:        stime,
+		End:          etime,
+		OutputDir:    outputDir,
+		EvalInterval: evalInterval,
+		URL:          url,
+	}
+	logger := log.NewLogfmtLogger(log.NewSyncWriter(os.Stderr))
+	ruleImporter := NewRuleImporter(logger, cfg)
+	err = ruleImporter.Init()
+	if err != nil {
+		fmt.Fprintln(os.Stderr, "rule importer init error", err)
+		return 1
+	}
+
+	errs := ruleImporter.LoadGroups(ctx, files)
+	for _, err := range errs {
+		if err != nil {
+			fmt.Fprintln(os.Stderr, "rule importer parse error", err)
+			return 1
+		}
+	}
+
+	errs = ruleImporter.ImportAll(ctx)
+	for _, err := range errs {
+		if err != nil {
+			fmt.Fprintln(os.Stderr, "rule importer error", err)
```

Comment:
Why doesn't this return with a non-zero exit code?

Reply:
will do.

Reply:
See my comment above

> It would be bit easier to return error here, but not a blocker.

To avoid this mistake in future.